pkg/otlp/internal/serializerexporter/factory.go

```diff
@@ -48,6 +49,8 @@ func (f *factory) createMetricExporter(ctx context.Context, params exp.CreateSet
 	exporter, err := exporterhelper.NewMetricsExporter(ctx, params, cfg, newExp.ConsumeMetrics,
 		exporterhelper.WithQueue(cfg.QueueSettings),
 		exporterhelper.WithTimeout(cfg.TimeoutSettings),
+		// the metrics remapping code mutates data
+		exporterhelper.WithCapabilities(consumer.Capabilities{MutatesData: true}),
```

Comment:
Can we do a recover in the agent so that otel component doesn't crash the whole agent ?

Reply:
Working on this in a separate PR since it can be done independently

Reply:
See #18640

Reply:
@liustanley It makes the exporter copy all OTLP data before running the consume functions, so that you can freely modify it without corrupting it for other exporters